### PR TITLE
E2E Test failure on CI #5766

### LINKED ISSUE #5766: E2E Test failure on CI
Catalog App E2E tests failed to run

> nx run catalog-e2e:e2e:ci 
- Generating browser application bundles...
✔ Browser application bundle generation complete.

Initial Chunk Files                                                                                              | Names                                                                                                         |      Size
vendor.js                                                                                                        | vendor                                                                                                        |   8.70 MB
main.js                                                                                                          | main                                                                                                          |   1.03 MB
polyfills-es5.js                                                                                                 | polyfills-es5                                                                                                 | 570.42 kB
styles.css                                                                                                       | styles                                                                                                        | 466.43 kB
scripts.js                                                                                                       | scripts                                                                                                       | 460.89 kB

[snip]

Build at: 2021-05-11T00:39:39.565Z - Hash: 8ae0b224c5366731aa3c - Time: 78138ms

Warning: /home/circleci/repo/apps/catalog/catalog/src/app/dashboard/right/list/list.component.ts is part of the TypeScript compilation but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **

✔ Compiled successfully.

Here E2E should have started but next app E2E tests are launched.

> nx run festival-e2e:e2e:ci   <============== Festival App E2E is run
- Generating browser application bundles...
✔ Browser application bundle generation complete.

Catalog App compiles fine but E2E tests don't run..

### PR DESCRIPTION
**Pull request by phpgeekfr**
fix #5766

### CODE REVIEW

**Comment by phpgeekfr** [comment]
@xmano  I added `|| true` to end of commands so CI can go to next step even if some tests are red, but this will make the tick green even if tests have failed ..

Example here: 
image


So we will need to look into logs to get results.

We will be able to remove `|| true` when the tests are green

**Comment by phpgeekfr** [comment]
@george43g do you know if there is way to continue to next step even if we have a red tick on some step  ? Instead of using `|| true`

**Comment by xmano** [comment]
> @xmano I added `|| true` to end of commands so CI can go to next step even if some tests are red, but this will make the tick green even if tests have failed ..
> 
> Example here:
> image
> 
> So we will need to look into logs to get results.
> 
> We will be able to remove `|| true` when the tests are green

In case of E2E tests, all the individual Apps are run even if there is a failure in some. So I don't understand what you are trying to do here.

The issue I am raising is of some App E2Es not running fully or getting aborted in between sometimes. This is causing tests to be skipped and incomplete results to be posted.

**Review by xmano — COMMENTED** [review]
Can you explain a bit about the changes?

Is there anything that can prevent tests from getting skipped?

**Comment by george43g** [comment]
> @george43g do you know if there is way to continue to next step even if we have a red tick on some step ? Instead of using `|| true`

Yes there's a way to fix this - I'll organise a fix tonight in the CI config

File: .circleci/config.yml [hunk 1]
```diff
@@ -337,6 +363,13 @@ workflows:
             branches:
               only:
                 - /upgrade\/.*/
+  run-e2e-tests:
+    jobs:
+      - run-e2e-tests:
```
Comment (george43g) [inline .circleci/config.yml:368]: we already do this in CI branch - why add it?

File: .circleci/config.yml [hunk 2]
```diff
@@ -277,6 +289,20 @@ jobs:
           no_output_timeout: 15m
       - save_build_cache
       - slack/status
+  run-e2e-tests:
+    executor: blockframes-ci
+    environment:
+      NG_CONFIG: ''
+    steps:
+      - prepare_environment
+      - run: echo "export PROJECT_ID=\"$(cut -d'/' -f2\<<<$CIRCLE_BRANCH)\"" >> $BASH_ENV
```
Comment (george43g) [inline .circleci/config.yml:298]: Is this to run e2e on any branch rather than CI? or against eny firebase project?

File: .circleci/config.yml [hunk 3]
```diff
@@ -143,7 +143,19 @@ commands:
...
         - run: npx cypress install
         - run:
-            command: npm run e2e
+            command: npx nx run festival-e2e:e2e:ci --headless || true
+            no_output_timeout: 20m
+        - run:
+            command: npx nx run financiers-e2e:e2e:ci --headless || true
+            no_output_timeout: 20m
+        - run:
+            command: npx nx run crm-e2e:e2e:ci --headless || true
```
Comment (george43g) [inline .circleci/config.yml:152]: I'll be able to find a better fix shortly, I think you can configure it to continue despite failure

**Review by george43g — CHANGES_REQUESTED** [review]

**Comment by phpgeekfr** [comment]
@xmano  
> So I don't understand what you are trying to do here.

this was a workaround to have every test on all app working all the time. I thought this is what you wanted

File: .circleci/config.yml (hunk 1 above)
Comment (phpgeekfr) [reply .circleci/config.yml:368]: So every user can run e2e on ci

File: .circleci/config.yml (hunk 2 above)
Comment (phpgeekfr) [reply .circleci/config.yml:298]: yes

**Comment by phpgeekfr** [comment]
Closing this one so @george43g and @xmano can find a better workaround

**Pull request closed without merging**